### PR TITLE
Remove soyjs dependency on soyhtml by giving soyjs it's own set of Pr…

### PR DESCRIPTION
**Pull request by mhupman**
…intDirectives that can be extended

### CODE REVIEW

File: soyjs/exec.go [hunk 1]
```diff
@@ -293,7 +292,7 @@ func (s *state) visitPrint(node *ast.PrintNode) {
 	s.indent()
 	s.js(s.bufferName, " += ")
 	for _, dir := range directives {
-		s.js("soy.$$", dir.Name, "(")
+		s.js(PrintDirectives[dir.Name].Name, "(")
```
Comment (robfig) [inline soyjs/exec.go]: this turns into a syntax error for |id or |noAutoescape, right?  I wonder if we can ignore them instead

Comment (robfig) [inline soyjs/exec.go]: oh, I see that's caught above

**Comment by robfig** [comment]
LGTM, thanks

File: soyjs/exec.go (hunk 1 above)
Comment (mhupman) [inline soyjs/exec.go]: Heh I thought that too at first.  Is it worth it to rename `directives` to `directiveNodes` or something to make it clearer that it is the template objects and not a collection of `PrintDirectives`?

**Pull request merged**